### DISCUSSION TITLE
Setting simulations parameters via a json file

### DISCUSSION THREAD

**Discussion by jpmeng** [jpmeng/MPLB#6] [Ideas]
Hi  @vtsinginos,

Could you please list issues that you find for the JSON input? Below is an initial list

*  Recognise parameters from OPS library. 
*  Consistency with the updated interface.

**Comment by vtsinginos** [comment]
Hi @jpmeng,

In the current version of OPS,  args to OPS are passed as args to the ops_init function, thus I believe it is not a good idea to add OPS args to JSON file.  

- Block information shall be given by block. For instance, the user gives externally the number of blocks and their names, and then he has to given in a structure all the block information, grid size, first grid node, boundary conditions, boundary type, node Types, etc, or even block connectivity, if we do not do it automatically.  Also, the information will be "concentrated" at a single location and a different user might be able to identify the parameters easier.
- Following this concept, it might be better to separate the variables per "action". We may categorise them as model parameters, i.e., relaxation times, simulations variables (number of steps, convergence rate, output steps), input variables etc.

I have uploaded the LBM-DEM model in the private fork. You will find it at the branch LBM-DEM. 
I will upload in few days the PSM on a branch of the public fork.
